contracts/core/EntryPoint.sol

```diff
@@ -521,6 +512,33 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuardT
         }
     }
 
+    // call sender.validateUserOp()
+    // handle wrong output size with FailedOp
+    function _callValidateUserOp(PackedUserOperation calldata op, UserOpInfo memory opInfo, uint256 missingAccountFunds, uint256 opIndex)
+    internal returns (uint256 validationData) {
+        uint256 saveFreePtr;
+        assembly ("memory-safe") {
+            saveFreePtr := mload(0x40)
+        }
+        bytes memory callData = abi.encodeCall(IAccount.validateUserOp, (op, opInfo.userOpHash, missingAccountFunds));
+        uint256 gasLimit = opInfo.mUserOp.verificationGasLimit;
+        address sender = opInfo.mUserOp.sender;
+        uint256 dataSize;
+        assembly ("memory-safe"){
+            let success := call(gasLimit, sender, 0, add(callData, 0x20), mload(callData), 0, 32)
+            dataSize := mul(returndatasize(), success)
+            validationData := mload(0)
+            mstore(0x40, saveFreePtr)
+        }
+        if (dataSize != 32) {
```

Comment:
```suggestion
        bool success;
        assembly ("memory-safe"){
            success := call(gasLimit, sender, 0, add(callData, 0x20), mload(callData), 0, 32)
            dataSize := returndatasize()
            validationData := mload(0)
            mstore(0x40, saveFreePtr)
        }
        if (!successs || dataSize != 32) {
```

Reply:
Maybe I don't understand the idea behind the `mul(returndatasize(), success)` but it seems like you could just check for `success`, which is a lot more reasonable?

Reply:
the logic is "ignore datasize in case of revert". I don't think that double-negative is more readable than multiplying "success" code by datasize. the both have the same effect. I also mload the returned data optimistically, and ignore it if datasize is not 32